src/wp-includes/class-wp-duotone.php

```diff
@@ -0,0 +1,1064 @@
+		// Functional syntax.
+		$is_match = preg_match(
+			'/^hsla?\(\s*([+-]?\d*\.?\d+)(deg|rad|grad|turn)?\s*,\s*([+-]?\d*\.?\d+)%\s*,\s*([+-]?\d*\.?\d+)%\s*(?:,\s*([+-]?\d*\.?\d+)(%)?\s*)?\)$/i',
+			$input,
+			$match
+		);
+
+		if ( ! $is_match ) {
+			// Whitespace syntax.
+			$is_match = preg_match(
+				'/^hsla?\(\s*([+-]?\d*\.?\d+)(deg|rad|grad|turn)?\s+([+-]?\d*\.?\d+)%\s+([+-]?\d*\.?\d+)%\s*(?:\/\s*([+-]?\d*\.?\d+)(%)?\s*)?\)$/i',
+				$input,
+				$match
+			);
+		}
+
+		if ( ! $is_match ) {
+			return null;
+		}
+
+		// For some reason, preg_match doesn't include empty matches at the end
+		// of the array, so we add them manually to make things easier later.
+		for ( $i = 1; $i <= 6; $i++ ) {
+			if ( ! isset( $match[ $i ] ) ) {
+				$match[ $i ] = '';
+			}
+		}
+
+		$hsla = self::colord_clamp_hsla(
+			array(
+				'h' => self::colord_parse_hue( $match[1], $match[2] ),
+				's' => (float) $match[3],
+				'l' => (float) $match[4],
+				'a' => '' === $match[5] ? 1 : (float) $match[5] / ( $match[6] ? 100 : 1 ),
+			)
+		);
+
+		return self::colord_hsla_to_rgba( $hsla );
+	}
+
+	/**
+	 * Direct port of colord's parse function simplified for our use case. This
+	 * version only supports string parsing and only returns RGBA values.
+	 *
+	 * @see https://github.com/omgovich/colord/blob/3f859e03b0ca622eb15480f611371a0f15c9427f/src/parse.ts#L37
+	 *
+	 * @param string $input The string to parse.
+	 * @return array|null An array of RGBA values or null if the string is invalid.
+	 */
+	private static function colord_parse( $input ) {
+		$result = self::colord_parse_hex( $input );
+
+		if ( ! $result ) {
+			$result = self::colord_parse_rgba_string( $input );
+		}
+
+		if ( ! $result ) {
+			$result = self::colord_parse_hsla_string( $input );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Take the inline CSS duotone variable from a block and return the slug. Handles styles slugs like:
+	 * var:preset|duotone|blue-orange
+	 * var(--wp--preset--duotone--blue-orange)
+	 *
+	 * @param string $duotone_attr The duotone attribute from a block.
+	 * @return string The slug of the duotone preset or an empty string if no slug is found.
+	 */
+	private static function get_slug_from_attribute( $duotone_attr ) {
+		// Uses Branch Reset Groups `(?|…)` to return one capture group.
+		preg_match( '/(?|var:preset\|duotone\|(\S+)|var\(--wp--preset--duotone--(\S+)\))/', $duotone_attr, $matches );
+
+		return ! empty( $matches[1] ) ? $matches[1] : '';
+	}
+
+	/**
+	 * Check if we have a valid duotone preset.
+	 *
+	 * @param string $duotone_attr The duotone attribute from a block.
+	 * @return bool True if the duotone preset present and valid.
+	 */
+	private static function is_preset( $duotone_attr ) {
+		$slug      = self::get_slug_from_attribute( $duotone_attr );
+		$filter_id = self::get_filter_id( $slug );
+
+		return array_key_exists( $filter_id, self::$global_styles_presets );
+	}
+
+	/**
+	 * Get the CSS variable name for a duotone preset.
```

Comment:
As per WP Documentation Standards, should be "Gets", not "Get" (third-person verb)
https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#language